src/java.desktop/share/classes/javax/swing/DefaultListSelectionModel.java

```diff
@@ -691,13 +691,18 @@ public void removeIndexInterval(int index0, int index1)
     {
         int rmMinIndex = Math.min(index0, index1);
         int rmMaxIndex = Math.max(index0, index1);
-        int gapLength = (rmMaxIndex - rmMinIndex) + 1;
+        int gapLength = ((rmMaxIndex - rmMinIndex) + 1) > (rmMaxIndex - rmMinIndex)
+                ? ((rmMaxIndex - rmMinIndex) + 1) : (rmMaxIndex - rmMinIndex);
```

Comment:
For negative values, the spec says "@throws IndexOutOfBoundsException if either index is less than {@code -1}" which it does.

Reply:
Well, the spec says nothing about it for this method: https://github.com/openjdk/jdk/blob/096bca4a9c5e8ac2668dd965df92153ea1d80add/src/java.desktop/share/classes/javax/swing/DefaultListSelectionModel.java#L684-L690

I guess it should be update to state it explicitly.

Reply:
To remove an index, I guess we need to first do setSelection where we will already throw IOOBE so I guess that's why this method does not mention or handle it explicitly

Reply:
> To remove an index, I guess we need to first do setSelection where we will already throw IOOBE so I guess that's why this method does not mention or handle it explicitly

It's not required by the implementation.

```java
        selectionModel.setSelectionInterval(0, 1);
        selectionModel.removeIndexInterval(Integer.MIN_VALUE, -1);
```

works … that is the call to `removeIndexInterval` throws the `IndexOutOfBoundsException` because it tries to copy the bits in the removed interval. If `rmMinIndex` is negative, the implementation may throw IOOBE before doing other work as an optimisation or rely on it being thrown from the code below as it does now.

Other methods specify the behaviour for negative parameters, this method would benefit from it too.

Reply:
I suggest amending `setSelectionInterval` to throw IOOBE if `Integer.MAX_VALUE` is passed.

Then I suggest amending `removeIndexInterval` to throw IOOBE if a negative value is passed or an index is `Integer.MAX_VALUE`. Explicit handling of negative values could be an optimisation made in addition to handling `MAX_VALUE`.

In both cases, the spec needs to be updated, and therefore the CSR will be required.

Reply:
I guess Integer.MAX_VALUE should be valid "int" value, its because of internal calculation which is causing it to overflow so I guess we should not update the spec for wrongful calculation. 
I have fixed the overflow in setSelectionInterval.

Reply:
This is not a generic class, its main purpose is to handle list selection in UI components, such as `JList`. Excluding `Integer.MAX_VALUE` makes sense to me, it simplifies the code: no additional check in the loop and no special handling `removeIndexInterval` which also adds a condition to the loop.

Reply:
OK. Practically, it seems right as list probably may not have MAX_VALUE entries..Updated to exclude MAX_VALUE and updated the javadoc.